src/app/admin/manageUser/ResetPasswordForm.tsx

```diff
@@ -10,78 +10,87 @@ import { UserRow } from "../usersTable/types";
     userToEdit: UserRow;
     onCancel: () => void;
     onConfirm: (alertOptions: AlertOptions) => void;
 }
 
-const ResetPasswordForm: React.FC<Props> = (props) => {
+interface UpdatePasswordResponse {
+    errorMessage: string | null;
+}
+
+const ErrorMessage = styled.span<{ theme: DefaultTheme }>`
+    color: ${(props) => props.theme.error};
+`;
+
+const ResetPasswordForm: React.FC<Props> = ({ userToEdit, onCancel, onConfirm }) => {
     const [password, setPassword] = useState("");
-    const passwordIsValid = checkPassword(password, userPasswordRules) === null;
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    const onConfirmPassword = async (): Promise<void> => {
+    const updatePassword = async (
+        userId: string,
+        newPassword: string
+    ): Promise<UpdatePasswordResponse> => {
         const response = await adminUpdateUserEmailAndPassword({
-            userId: props.userToEdit.userId,
-            attributes: { password },
+            userId: userId,
+            attributes: { password: newPassword },
         });
 
-        if (response.error === null) {
-            props.onConfirm({
-                success: true,
-                message: (
-                    <>
-                        Password for <b>{props.userToEdit.email}</b> updated successfully.
-                    </>
-                ),
-            });
-            void logInfoReturnLogId(`Password for ${props.userToEdit.email} updated successfully`);
-        } else {
-            const logId = await logErrorReturnLogId(
-                `Error resetting password userId: ${props.userToEdit.userId}`
-            );
-            props.onConfirm({
-                success: false,
-                message: <>Reset password operation failed. Log ID: {logId}</>,
+        if (response.error) {
+            void logErrorReturnLogId(`Error resetting password userId: ${userId}`, {
+                response,
             });
+            return { errorMessage: response.error["Failed to update user"] };
         }
+        return { errorMessage: null };
+    };
+
+    const onConfirmPassword = (): void => {
+        updatePassword(userToEdit.userId, password).then(({ errorMessage }) => {
+            console.log(errorMessage);
+            setErrorMessage(errorMessage);
+            if (errorMessage) {
+                setErrorMessage(errorMessage);
+            } else {
+                onConfirm({
+                    success: true,
+                    message: (
+                        <>
+                            Password for <b>{userToEdit.email}</b> updated successfully.
+                        </>
+                    ),
+                });
+                void logInfoReturnLogId(`Password for ${userToEdit.email} updated successfully`);
+            }
+        });
+    };
+
+    const safeSubmit = (event: React.MouseEvent<HTMLButtonElement>): void => {
+        event.preventDefault();
+        onConfirmPassword();
```

Comment:
I think that rather than wrap this we can just directly make `onConfirmPassword` take the event as an argument?